src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -0,0 +1,272 @@
+import {
+  ExpectedVersionConflictError,
+  STREAM_DOES_NOT_EXIST,
+  assertExpectedVersionMatchesCurrent,
+  type EventStore,
+  type Event,
+  type AggregateStreamOptions,
+  type AggregateStreamResult,
+  type AppendToStreamOptions,
+  type AppendToStreamResult,
+  type ReadStreamOptions,
+  type ReadStreamResult,
+  type ReadEvent,
+  type ReadEventMetadata,
+  type ExpectedStreamVersion,
+} from '@event-driven-io/emmett';
+import { type Collection, MongoClient, type WithId } from 'mongodb';
+import { v4 as uuid } from 'uuid';
+
+export const MongoDBEventStoreDefaultStreamVersion = -1;
+export const MongoDBDefaultCollectionName = 'eventstreams';
```

Comment:
**SUGGESTION:** Something to do for the future would be to check if we should use a single collection for all streams, or break it down by stream type. It can stay like that, but later on some performance and load test would be needed to ensure that won't slow down the performance.

Reply:
Playing around with multiple collections separated by stream type, should we rename the `streamName` property on `EventStream` to just be the `streamId` instead of the fully formatted name including the stream type? Figured if each collection is the type, we can just use the streamId itself for each doc